docs/frontend-system/building-plugins/06-swappable-components.md

```diff
@@ -0,0 +1,98 @@
+---
+id: swappable-components
+title: Swappable components
+sidebar_label: Swappable components
+description: Configuring or overriding Swappable Components
+---
+
+# Swappable components
+
+Swappable components are a feature of the frontend system that allow you to replace the implementations of components that are used in your Backstage app.
+These Swappable Components are defined using `createSwappableComponent` and then can be exported from a plugins `-react` package in order to be used in both other plugins, and to be rebound to a new implementation by the Backstage Integrator.
+
+## Creating a Swappable Component
+
+In order to create a Swappable Component, you need to use the `createSwappableComponent` function from the `@backstage/frontend-plugin-api` package. You can supply a default implementation for the component, as well as a way to separate both the props of the external component and in the implementation of the component.
+
+```tsx title="in @internal/plugin-example-react"
+import { createSwappableComponent } from '@backstage/frontend-plugin-api';
+
+export const ExampleSwappableComponent = createSwappableComponent({
+  name: 'example',
+
+  // This is a loader for loading the default implementation of the component when there's no overriden
+  // implementation created with `SwappableComponentBlueprint`.
+  // It can be sync like below, but is can also be async like `loader: () => import('./DefaultImplementation').then(m => m.default)`.
```

Comment:
```suggestion
  // It can be sync like below, but is can also be async like `loader: () => import('./DefaultImplementation').then(m => m.DefaultImplementation)`.
```

Would typically be what we recommend